diff --git a/node_modules/react-native-youtube-iframe/lib/commonjs/YoutubeIframe.js b/node_modules/react-native-youtube-iframe/lib/commonjs/YoutubeIframe.js
index 0be2dad..bfd830f 100644
--- a/node_modules/react-native-youtube-iframe/lib/commonjs/YoutubeIframe.js
+++ b/node_modules/react-native-youtube-iframe/lib/commonjs/YoutubeIframe.js
@@ -101,15 +101,22 @@ const YoutubeIframe = (props, ref) => {
     },
     seekTo: (seconds, allowSeekAhead) => {
       webViewRef.current.injectJavaScript(_PlayerScripts.PLAYER_FUNCTIONS.seekToScript(seconds, allowSeekAhead));
+    },
+    playVideo: () => {
+      webViewRef.current?.injectJavaScript(_PlayerScripts.PLAYER_FUNCTIONS.playVideo);
+    },
+    pauseVideo: () => {
+      webViewRef.current?.injectJavaScript(_PlayerScripts.PLAYER_FUNCTIONS.pauseVideo);
     }
   }), []);
   (0, _react.useEffect)(() => {
+    if (!playerReady) return;
     if (play) {
-      sendPostMessage('playVideo', {});
+      webViewRef.current?.injectJavaScript(_PlayerScripts.PLAYER_FUNCTIONS.playVideo);
     } else {
-      sendPostMessage('pauseVideo', {});
+      webViewRef.current?.injectJavaScript(_PlayerScripts.PLAYER_FUNCTIONS.pauseVideo);
     }
-  }, [play, sendPostMessage]);
+  }, [play, playerReady]);
   (0, _react.useEffect)(() => {
     if (mute) {
       sendPostMessage('muteVideo', {});
diff --git a/node_modules/react-native-youtube-iframe/src/YoutubeIframe.js b/node_modules/react-native-youtube-iframe/src/YoutubeIframe.js
index 4bbf487..26bf47e 100644
--- a/node_modules/react-native-youtube-iframe/src/YoutubeIframe.js
+++ b/node_modules/react-native-youtube-iframe/src/YoutubeIframe.js
@@ -121,17 +121,24 @@ const YoutubeIframe = (props, ref) => {
           PLAYER_FUNCTIONS.seekToScript(seconds, allowSeekAhead),
         );
       },
+      playVideo: () => {
+        webViewRef.current?.injectJavaScript(PLAYER_FUNCTIONS.playVideo);
+      },
+      pauseVideo: () => {
+        webViewRef.current?.injectJavaScript(PLAYER_FUNCTIONS.pauseVideo);
+      },
     }),
     [],
   );
 
   useEffect(() => {
+    if (!playerReady) return;
     if (play) {
-      sendPostMessage('playVideo', {});
+      webViewRef.current?.injectJavaScript(PLAYER_FUNCTIONS.playVideo);
     } else {
-      sendPostMessage('pauseVideo', {});
+      webViewRef.current?.injectJavaScript(PLAYER_FUNCTIONS.pauseVideo);
     }
-  }, [play, sendPostMessage]);
+  }, [play, playerReady]);
 
   useEffect(() => {
     if (mute) {
